### PR TITLE
deprecate type()

### PR DESCRIPTION
**Pull request by Phil-Friderici**

### CODE REVIEW

File: spec/classes/init_spec.rb [hunk 1]
```diff
@@ -415,7 +415,7 @@
       it do
         expect {
           should contain_class('ntp')
-        }.to raise_error(Puppet::Error, /ntp::peers must be a string or an array or an hash. Detected type is <boolean>./)
+        }.to raise_error(Puppet::Error, /ntp::peers must be a string or an array or an hash./)
```
Comment (Phil-Friderici) [inline spec/classes/init_spec.rb]: we could add detecion for other (unsupported) types as well, but I don't think that's worth the effort here.

**Comment by Phil-Friderici** [comment]
@ghoneycutt There will be another PR to prepare the array handling in the template file.

File: manifests/init.pp [hunk 2]
```diff
@@ -39,61 +39,56 @@
...
+      $my_peers = $peers
+      validate_hash($my_peers)
+    }
+    elsif is_string($peers) == true {
+      $peers_type = 'string'
+      $my_peers = any2array($peers)
+      validate_array($my_peers)
+    }
+    else {
+      fail("ntp::peers must be a string or an array or an hash.")
```
Comment (Phil-Friderici) [inline manifests/init.pp]: we could add detecion for other (unsupported) types as well, but I don't think that's worth the effort here.

File: spec/classes/init_spec.rb (hunk 1 above)
Comment (ghoneycutt) [inline spec/classes/init_spec.rb]: agreed

File: manifests/init.pp (hunk 2 above)
Comment (ghoneycutt) [inline manifests/init.pp]: `a hash` not `an hash`.

**Comment by ghoneycutt** [comment]
Released in v3.5.1

Thanks @Phil-Friderici !

**Pull request merged**